cmd/skopeo/utils.go

```diff
@@ -16,8 +17,18 @@ type errorShouldDisplayUsage struct {
 	error
 }
 
+// Check if container-storage are used before doing the maybeReexec
 func needsRexec(c *cli.Context) error {
-	return maybeReexec()
+	for _, arg := range c.Args() {
```

Comment:
This is _way_ too imprecise.  Instead:
- In the various commands, drop `Before: needsReexec`
- In each of the command handlers, run something like `maybeReexec(inputParameter)` _only for the relevant parameters_, something like:
    ```go
        inputImageName := image.args[0]
        if err := maybeReexec(inputImageName); err != nil { 
           return err
        }
    	… := parseImageSource(ctx, opts.image, inputImage) // or parseImage or alltransports.ParseImageName
    ```
    possibly rearranging this so that the `maybeReexec`s happen at the start of the command handlers, before any other real work gets done.